daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala

```diff
@@ -1844,12 +1944,99 @@ object VerifyTestCase {
    */
-  def stripLocationInformation(msg: String): String = {
-    msg.replaceAll("Location line.*", "")
+  def verifyAnyOfDiagnosticsFound(
+    actualDiags: Seq[Diagnostic],
+    expectedDiags: Option[ErrorWarningBase],
+    implString: Option[String],
+  ) = {
+
+    val actualDiagMsgs = actualDiags.map {
+      _.toString()
+    }
+    val expectedDiagMsgs = expectedDiags
+      .map {
+        _.messages
+      }
+      .getOrElse(Nil)
+
+    if (expectedDiags.isDefined && actualDiags.isEmpty) {
+      throw TDMLException(
+        """"Diagnostic message(s) were expected but not found."""" +
+          "\n" +
+          """Expected: """ + expectedDiagMsgs.mkString("\n") +
+          (if (actualDiagMsgs.isEmpty)
+             "\n No diagnostic messages were issued."
+           else
+             "\n The actual diagnostics messages were: " + actualDiagMsgs.mkString("\n")),
+        implString,
+      )
+    }
+
+    // must find at least one expected warning message within some
+    // actual warning message.
+    val wasFound = expectedDiags.exists { expectedDiag =>
+      expectedDiag.messages.exists { expectedMsg =>
+        val wasFound = actualDiags.exists { actualDiag =>
+          val actualDiagMsg = actualDiag.getMessage()
+          actualDiag.isError == expectedDiag.isError &&
+          actualDiagMsg.toLowerCase.contains(expectedMsg.toLowerCase)
+        }
+        wasFound
+      }
+    }
+    if (!wasFound) {
+      throw TDMLException(
+        s"""Did not find any of the diagnostic ${expectedDiags.head.diagnosticType} message """" +
+          expectedDiagMsgs.mkString("\n") +
+          """" in any of the actual diagnostic messages: """ + "\n" +
+          actualDiagMsgs.mkString("\n"),
+        implString,
+      )
+    }
+  }
+
+  def verifyNoneOfDiagnosticsFound(
+    actualDiags: Seq[Diagnostic],
+    expectedDiags: Option[ErrorWarningBase],
+    implString: Option[String],
+  ): Unit = {
+
+    val actualDiagMsgs = actualDiags.map {
+      _.toString()
+    }
+    val expectedDiagMsgs = expectedDiags
+      .map {
+        _.messages
+      }
+      .getOrElse(Nil)
+
+    // must not find any of the expected warning message within some
+    // actual warning message.
+    val wasFound = expectedDiags.exists { expectedDiag =>
+      expectedDiag.messages.exists { expectedMsg =>
+        val wasFound = actualDiags.exists { actualDiag =>
+          val actualDiagMsg = actualDiag.getMessage()
+          actualDiag.isError == expectedDiag.isError &&
+          actualDiagMsg.toLowerCase.contains(expectedMsg.toLowerCase)
+        }
+        wasFound
+      }
+    }
+    if (wasFound) {
+      throw TDMLException(
+        s"""Found one of diagnostic ${expectedDiags.head.diagnosticType} message """" +
+          expectedDiagMsgs.mkString("\n") +
+          """" in some of the actual diagnostic messages: """ + "\n" +
+          actualDiagMsgs.mkString("\n"),
+        implString,
+      )
+    }
```

Comment:
It might be useful to reduce duplication by combining these functions into a single one that accepts a mode. It feels like are the core they are pretty similar just with some minor differences. For example, you could maybe do something like:

```scala
val diagsFound = expectedDiags.map { diag => 
  val wasFound = actualDiags.contains(diag)
  (diag, wasFound)
}

matchAttr match {
  case "all" if (!diagsFound.forall(_._2)) => {
     throw TDMLException(...)
  }
  case "any" if (!diagsFound.exists(_._2)) => {
     throw TDMLException(...)
  }
  case "none" if (diagsFound.exists(_._2)) => {
    throw TDMLException(...)
  }
}
```

So the logic to determine which diags are found is exactly the same with out copy/paste, and we create a Seq that determines which diags are found/not found. Then a match/case does the logic to figure out which of those were failures depending on the match mode and to create the appropriate exception.